examples/preprocessing/plot_muscle_detection.py

```diff
@@ -0,0 +1,85 @@
+"""
+===========================
+Annotate muscle artifacts
+===========================
+
+Muscle contractions produce high frequency activity that can contaminate the
+brain signal of interest. Muscle artifacts can be produced when clinching the
+jaw, swalloing, or twitching a head muscle. Muscle artifacts are most notable
+in the range of 110-140Hz.
+
+This example uses `annotate_muscle` to annotate segments where muscle activity
+likely occurred. This is done by band-pass filtering the data in the 110-140 Hz
+range. Then, the envelope is taken to account for peaks and troughs. The
+envelope is z-scored and averaged across channels. To remove noisy transient
+peaks, the z-scored average is low-pass filtered to 4 Hz. Segments above a set
```

Comment:
> To remove noisy transient peaks...

This whole procedure (bandpass the typical range of muscle activity, extract the envelope, z-score, and now low-passing) is about removing noise. Can you explain this low-passing step as something other than "removing noise"?  What is it about those transient peaks that make them necessary to remove?  Are they brain signal that we don't want accidentally to get rid of?  Are they muscle artifacts, but the transient nature of them makes them hard to deal with for some reason? Are they just too short to be worth annotating?

Reply:
I'm also not sure whether this step is necessary. @AdoNunes - do you have some comparisons that show the benefit of low-passing the high-frequency envelope?

Reply:
I checked it a few months ago, this avoids very small artifact annotations, also, this smoothing increases the time window of the annotation and better captures the start or end of the muscle activity which is not as strong. I still have to look for a good dataset with muscle artifacts.

Reply:
I'm ok with very short annotations - you can just change the z threshold and reduce their occurrence. To better capture start and end you could just extend the annotation by +- some s (could be a parameter), I think this is how it is done in fieldtrip.

Reply:
As I user I prefer it like this, I found it to be much better, I didn't have to go over the data and unmark annotation lines that are just false positives, moreover, just extending the annotation by a fixed length doesn't mean that it will all be muscle.

Reply:
would it be a good compromise to have `min_length_bad` and `min_length_good` parameters?  The first would set a lower bound on muscle artifacts, and the second would set a lower bound on good data spans in between muscle artifacts.

As for expanding the annotation edges in time, I think this should not be automatic. Either there should be a parameter, or users should be left to programmatically lengthen the annotations after creation (or manually do it by click-dragging annotation borders).

Reply:
I continued the discussion in the main conversation thread